File: 3078/3078.py
class Solution:
    def match_alphanumerical_pattern_in_matrix_i(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0
        return 1 + max(self.match_alphanumerical_pattern_in_matrix_i(root.left), self.match_alphanumerical_pattern_in_matrix_i(root.right))

File: 3078/3078.go
func matchAlphanumericalPatternInMa(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := matchAlphanumericalPatternInMa(root.Left)
	r := matchAlphanumericalPatternInMa(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}